config

```diff
@@ -1,532 +1,14 @@
-ngx_feature_path=
-ngx_feature_test='printf("hello");'
-
-. auto/feature
-
-if [ $ngx_found = yes ]; then
-    CORE_LIBS="-Wl,-E $CORE_LIBS"
-fi
-
-# for Cygwin
-ngx_feature="export symbols by default (--export-all-symbols)"
-ngx_feature_libs="-Wl,--export-all-symbols"
-ngx_feature_name=
-ngx_feature_run=no
-ngx_feature_incs="#include <stdio.h>"
-ngx_feature_path=
-ngx_feature_test='printf("hello");'
-
-. auto/feature
-
-if [ $ngx_found = yes ]; then
-    CORE_LIBS="-Wl,--export-all-symbols $CORE_LIBS"
-fi
-
-NGX_DTRACE_PROVIDERS="$NGX_DTRACE_PROVIDERS $ngx_addon_dir/dtrace/ngx_lua_provider.d"
-NGX_TAPSET_SRCS="$NGX_TAPSET_SRCS $ngx_addon_dir/tapset/ngx_lua.stp"
-
-USE_MD5=YES
-USE_SHA1=YES
-
-CORE_INCS="$CORE_INCS $ngx_addon_dir/src/api"
-
-ngx_feature="SO_PASSCRED"
-ngx_feature_libs=
-ngx_feature_name="NGX_HTTP_LUA_HAVE_SO_PASSCRED"
-ngx_feature_run=no
-ngx_feature_incs="#include <sys/types.h>
-#include <sys/socket.h>"
-ngx_feature_path=
-ngx_feature_test='setsockopt(1, SOL_SOCKET, SO_PASSCRED, NULL, 0);'
-
-. auto/feature
-
-ngx_feature="__attribute__(constructor)"
-ngx_feature_libs=
-ngx_feature_name="NGX_HTTP_LUA_HAVE_CONSTRUCTOR"
-ngx_feature_run=yes
-ngx_feature_incs="#include <stdlib.h>
-int a = 2;
-__attribute__((constructor))
-static void foo(void)
-{
-    a = 0;
-}
-"
-ngx_feature_test="exit(a);"
-SAVED_CC_TEST_FLAGS="$CC_TEST_FLAGS"
-CC_TEST_FLAGS="-Werror -Wall $CC_TEST_FLAGS"
-
-. auto/feature
-
-CC_TEST_FLAGS="$SAVED_CC_TEST_FLAGS"
-
-# ----------------------------------------
-
-ngx_feature="malloc_trim"
-ngx_feature_libs=
-ngx_feature_name="NGX_HTTP_LUA_HAVE_MALLOC_TRIM"
-ngx_feature_run=yes
-ngx_feature_incs="#include <malloc.h>
-#include <stdio.h>"
-ngx_feature_test="int rc = malloc_trim((size_t) 0); printf(\"%d\", rc);"
-SAVED_CC_TEST_FLAGS="$CC_TEST_FLAGS"
-CC_TEST_FLAGS="-Werror -Wall $CC_TEST_FLAGS"
-
-. auto/feature
-
-CC_TEST_FLAGS="$SAVED_CC_TEST_FLAGS"
-
-# ----------------------------------------
-
-if test -n "$ngx_module_link"; then
-    ngx_module_type=HTTP_AUX_FILTER
-    ngx_module_name=$ngx_addon_name
-    ngx_module_deps="$HTTP_LUA_DEPS"
-    ngx_module_srcs="$HTTP_LUA_SRCS"
-
-    . auto/module
-else
-    HTTP_AUX_FILTER_MODULES="$HTTP_AUX_FILTER_MODULES $ngx_addon_name"
-    NGX_ADDON_SRCS="$NGX_ADDON_SRCS $HTTP_LUA_SRCS"
-    NGX_ADDON_DEPS="$NGX_ADDON_DEPS $HTTP_LUA_DEPS"
-
-    CORE_INCS="$CORE_INCS $ngx_module_incs"
-    CORE_LIBS="$CORE_LIBS $ngx_module_libs"
-fi
-
-#CFLAGS=$"$CFLAGS -DLUA_DEFAULT_PATH='\"/usr/local/openresty/lualib/?.lua\"'"
-#CFLAGS=$"$CFLAGS -DLUA_DEFAULT_CPATH='\"/usr/local/openresty/lualib/?.so\"'"
+case $NGX_CC_NAME in
+    gcc | clang)
+        echo "using gcc-specific config"
+        . $ngx_addon_dir/auto/config.gcc;
+    ;;
+    msvc)
+        echo "using msvc-specific config"
+        . $ngx_addon_dir/auto/config.msvc;
+    ;;
+    *)
+        echo "using untested compiler to build the module";
+        . $ngx_addon_dir/auto/config.gcc;
+    ;;
+esac
```

Comment:
We need to add back the following commented lines at the end of this file. They are meaningful to OpenResty's build toolchain:

```bash
#CFLAGS=$"$CFLAGS -DLUA_DEFAULT_PATH='\"/usr/local/openresty/lualib/?.lua\"'"
#CFLAGS=$"$CFLAGS -DLUA_DEFAULT_CPATH='\"/usr/local/openresty/lualib/?.so\"'"
```

Reply:
Done.